src/dfsane.jl

```diff
@@ -0,0 +1,314 @@
+                                            nₑₓₚ,
+                                            ηₛ,
+                                            max_inner_iterations)
+end
+mutable struct DFSaneCache{iip, fType, algType, uType, resType, T, pType,
+                           INType,
+                           tolType,
+                           probType}
+    f::fType
+    alg::algType
+    uₙ::uType
+    uₙ₋₁::uType
+    fuₙ::resType
+    fuₙ₋₁::resType
+    𝒹::uType
+    ℋ::uType
+    f₍ₙₒᵣₘ₎ₙ₋₁::T
+    f₍ₙₒᵣₘ₎₀::T
+    M::Int
+    σₙ::T
+    σₘᵢₙ::T
+    σₘₐₓ::T
+    α₁::T
+    γ::T
+    τₘᵢₙ::T
+    τₘₐₓ::T
+    nₑₓₚ::Int
+    p::pType
+    force_stop::Bool
+    maxiters::Int
+    internalnorm::INType
+    retcode::SciMLBase.ReturnCode.T
+    abstol::tolType
+    prob::probType
+    stats::NLStats
+    function DFSaneCache{iip}(f::fType, alg::algType, uₙ::uType, uₙ₋₁::uType,
+                              fuₙ::resType, fuₙ₋₁::resType, 𝒹::uType, ℋ::uType,
+                              f₍ₙₒᵣₘ₎ₙ₋₁::T, f₍ₙₒᵣₘ₎₀::T, M::Int, σₙ::T, σₘᵢₙ::T, σₘₐₓ::T,
+                              α₁::T, γ::T, τₘᵢₙ::T, τₘₐₓ::T, nₑₓₚ::Int, p::pType,
+                              force_stop::Bool, maxiters::Int, internalnorm::INType,
+                              retcode::SciMLBase.ReturnCode.T, abstol::tolType,
+                              prob::probType,
+                              stats::NLStats) where {iip, fType, algType, uType,
+                                                     resType, T, pType, INType,
+                                                     tolType,
+                                                     probType
+                                                     }
+        new{iip, fType, algType, uType, resType, T, pType, INType, tolType,
+            probType
+            }(f, alg, uₙ, uₙ₋₁, fuₙ, fuₙ₋₁, 𝒹, ℋ, f₍ₙₒᵣₘ₎ₙ₋₁, f₍ₙₒᵣₘ₎₀, M, σₙ,
+              σₘᵢₙ, σₘₐₓ, α₁, γ, τₘᵢₙ,
+              τₘₐₓ, nₑₓₚ, p, force_stop, maxiters, internalnorm,
+              retcode,
+              abstol, prob, stats)
+    end
+end
+
+function SciMLBase.__init(prob::NonlinearProblem{uType, iip}, alg::DFSane,
+                          args...;
+                          alias_u0 = false,
+                          maxiters = 1000,
+                          abstol = 1e-6,
+                          internalnorm = DEFAULT_NORM,
+                          kwargs...) where {uType, iip}
+    if alias_u0
+        uₙ = prob.u0
+    else
+        uₙ = deepcopy(prob.u0)
+    end
+
+    p = prob.p
+    T = eltype(uₙ)
+    σₘᵢₙ, σₘₐₓ, γ, τₘᵢₙ, τₘₐₓ = T(alg.σₘᵢₙ), T(alg.σₘₐₓ), T(alg.γ), T(alg.τₘᵢₙ), T(alg.τₘₐₓ)
+    α₁ = one(T)
+    γ = T(alg.γ)
+    f₍ₙₒᵣₘ₎ₙ₋₁ = α₁
+    σₙ = T(alg.σ₁)
+    M = alg.M
+    nₑₓₚ = alg.nₑₓₚ
+    𝒹, uₙ₋₁, fuₙ, fuₙ₋₁ = copy(uₙ), copy(uₙ), copy(uₙ), copy(uₙ)
+
+    if iip
+        f(dx, x) = prob.f(dx, x, p)
+        f(fuₙ₋₁, uₙ₋₁)
+
+    else
+        f(x) = prob.f(x, p)
+        fuₙ₋₁ = f(uₙ₋₁)
```

Comment:
The error is because `f` is being overwritten due to branching. Changing the name might fix it.

Reply:
Changing the name does sort of fix it. But I'm sorry, I really don't understand why this happens. I never had this problem before. Do you have some quick reference?

Reply:
You just cannot define standard functions in branches, make them anonymous.

Reply:
Thanks!